src/main/java/uk/gov/companieshouse/api/strikeoffobjections/authorization/UserAuthorizationInterceptor.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.api.strikeoffobjections.authorization;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IObjectionService;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderFields;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderParser;
+import uk.gov.companieshouse.service.ServiceException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserAuthorizationInterceptor implements HandlerInterceptor {
+
+    private final ApiLogger apiLogger;
+    private final  IObjectionService objectionService;
+    private final ERICHeaderParser ericHeaderParser;
+
+    public UserAuthorizationInterceptor(ApiLogger apiLogger, IObjectionService objectionService, ERICHeaderParser ericHeaderParser) {
+        this.apiLogger = apiLogger;
+        this.objectionService = objectionService;
+        this.ericHeaderParser = ericHeaderParser;
+
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        final String requestId = request.getHeader(ERICHeaderFields.ERIC_REQUEST_ID);
+        apiLogger.debugContext(requestId, "Checking current user is authorised for objection id");
+        final String user = request.getHeader(ERICHeaderFields.ERIC_AUTHORISED_USER);
+        final String objectionId = getObjectionId(request.getRequestURI());
+
+        final Objection objection = objectionService.getObjection(requestId, objectionId);
+
+        final String createdByUserEmail = objection.getCreatedBy().getEmail();
+        final String requestUserEmail = ericHeaderParser.getEmailAddress(user);
+
+        if(createdByUserEmail.equals(requestUserEmail)) {
```

Comment:
Not sure this will allow a dissolution team member to download an attachment related to an objection that was created by someone else.

Reply:
I think you're right, i'll make sure thats addressed

Reply:
also thinking we should allow a support team person to re-run failures via patch. Dn't now if there's a requirement for it yet though.